pgtap/dijkstra/dijkstraVia-rows-check.sql

```diff
@@ -0,0 +1,349 @@
+SELECT * FROM pgr_dijkstraVia(
+    'SELECT id, source, target, cost, reverse_cost
+    FROM edge_table
+    ORDER BY id',
+    ARRAY[7, 3, 2, 12, 7],
+    directed => false
+);
+
+PREPARE descendingOrderUndirected1 AS
+SELECT * FROM pgr_dijkstraVia(
+    'SELECT id, source, target, cost, reverse_cost
+    FROM edge_table
+    ORDER BY id DESC',
+    ARRAY[7, 3, 2, 12, 7],
+    directed => false
+);
+
+PREPARE randomOrderUndirected1 AS
+SELECT * FROM pgr_dijkstraVia(
+    'SELECT id, source, target, cost, reverse_cost
+    FROM edge_table
+    ORDER BY RANDOM()',
+    ARRAY[7, 3, 2, 12, 7],
+    directed => false
+);
+
+-- Not strict, Not Allowed U turn
+
+PREPARE expectedOutputUndirected2 AS
+SELECT * FROM pgr_dijkstraVia(
+    'SELECT id, source, target, cost, reverse_cost
+    FROM edge_table
+    ORDER BY id',
+    ARRAY[7, 3, 2, 12, 7], U_turn_on_edge => false,
+    directed => false
+);
+
+PREPARE descendingOrderUndirected2 AS
+SELECT * FROM pgr_dijkstraVia(
+    'SELECT id, source, target, cost, reverse_cost
+    FROM edge_table
+    ORDER BY id DESC',
+    ARRAY[7, 3, 2, 12, 7], U_turn_on_edge => false,
+    directed => false
+);
+
+PREPARE randomOrderUndirected2 AS
+SELECT * FROM pgr_dijkstraVia(
+    'SELECT id, source, target, cost, reverse_cost
+    FROM edge_table
+    ORDER BY RANDOM()',
+    ARRAY[7, 3, 2, 12, 7], U_turn_on_edge => false,
+    directed => false
+);
+
+-- Strict, Allowed U turn
+
+PREPARE expectedOutputUndirected3 AS
+SELECT * FROM pgr_dijkstraVia(
+    'SELECT id, source, target, cost, reverse_cost
+    FROM edge_table
+    ORDER BY id',
+    ARRAY[7, 3, 2, 12, 7], strict => false,
+    directed => false
+);
+
+PREPARE descendingOrderUndirected3 AS
+SELECT * FROM pgr_dijkstraVia(
+    'SELECT id, source, target, cost, reverse_cost
+    FROM edge_table
+    ORDER BY id DESC',
+    ARRAY[7, 3, 2, 12, 7], strict => false,
+    directed => false
+);
+
+PREPARE randomOrderUndirected3 AS
+SELECT * FROM pgr_dijkstraVia(
+    'SELECT id, source, target, cost, reverse_cost
+    FROM edge_table
+    ORDER BY RANDOM()',
+    ARRAY[7, 3, 2, 12, 7], strict => false,
+    directed => false
+);
+
+-- Strict, Not allowed U turn
+
+PREPARE expectedOutputUndirected4 AS
+SELECT * FROM pgr_dijkstraVia(
+    'SELECT id, source, target, cost, reverse_cost
+    FROM edge_table
+    ORDER BY id',
+    ARRAY[7, 3, 2, 12, 7], strict => false, U_turn_on_edge => false,
+    directed => false
+);
+
+PREPARE descendingOrderUndirected4 AS
+SELECT * FROM pgr_dijkstraVia(
+    'SELECT id, source, target, cost, reverse_cost
+    FROM edge_table
+    ORDER BY id DESC',
+    ARRAY[7, 3, 2, 12, 7], strict => false, U_turn_on_edge => false,
+    directed => false
+);
+
+PREPARE randomOrderUndirected4 AS
+SELECT * FROM pgr_dijkstraVia(
+    'SELECT id, source, target, cost, reverse_cost
+    FROM edge_table
+    ORDER BY RANDOM()',
+    ARRAY[7, 3, 2, 12, 7], strict => false, U_turn_on_edge => false,
+    directed => false
+);
+
+-- Directed
+
```

Comment:
Wrap all the tests of dijkstraVia with a TODO
otherwise we will never see again the server crash.